shotover-proxy/src/transforms/cassandra/sink_cluster.rs

```diff
@@ -0,0 +1,165 @@
+use super::connection::CassandraConnection;
+use crate::codec::cassandra::CassandraCodec;
+use crate::concurrency::FuturesOrdered;
+use crate::error::ChainResponse;
+use crate::frame::cassandra;
+use crate::message::Messages;
+use crate::tls::TlsConfig;
+use crate::tls::TlsConnector;
+use crate::transforms::util::Response;
+use crate::transforms::{Transform, Transforms, Wrapper};
+use anyhow::Result;
+use async_trait::async_trait;
+use cassandra_protocol::frame::Opcode;
+use metrics::{register_counter, Counter};
+use serde::Deserialize;
+use std::time::Duration;
+use tokio::sync::{mpsc, oneshot};
+use tokio::time::timeout;
+use tokio_stream::StreamExt;
+use tracing::{error, trace};
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct CassandraSinkClusterConfig {
+    pub first_contact_points: Vec<String>,
+    pub data_center: String,
+    pub tls: Option<TlsConfig>,
+}
+
+impl CassandraSinkClusterConfig {
+    pub async fn get_transform(&self, chain_name: String) -> Result<Transforms> {
+        let tls = self.tls.clone().map(TlsConnector::new).transpose()?;
+        Ok(Transforms::CassandraSinkCluster(CassandraSinkCluster::new(
+            self.first_contact_points.clone(),
+            chain_name,
+            tls,
+        )))
+    }
+}
+
+pub struct CassandraSinkCluster {
+    contact_points: Vec<String>,
+    outbound: Option<CassandraConnection>,
+    chain_name: String,
+    failed_requests: Counter,
+    tls: Option<TlsConnector>,
+    pushed_messages_tx: Option<mpsc::UnboundedSender<Messages>>,
+}
+
+impl Clone for CassandraSinkCluster {
```

Comment:
Why do we need this?

Reply:
We need a custom clone for the same reason as for CassandraSinkSingle.
we need to not clone `self.outbound` so that the new transform creates its own outbound connection.